pkg/generator_test.go

```diff
@@ -240,7 +240,7 @@ func NewRequester(t testing.TB) *Requester {
 }
 
 func (s *GeneratorSuite) TestGeneratorExpecterComplete() {
-	expectedBytes, err := ioutil.ReadFile(getFixturePath("mocks", "expecter.go"))
+	expectedBytes, err := ioutil.ReadFile(getMocksPath("ExpecterTest.go"))
```

Comment:
The thing I agree with @Gevrai here is that it's odd to be asserting the output of mockery with the output of mockery (you see what I mean?). You are generating `ExepcterTest.go` in the `mocks` package, then you generate it again in this test and assert that it's the same. It's not _necessarily_ wrong to do it this way, but none of the other tests do it this way (see `TestGeneratorFunction` below). 

Although, I can see an argument to be made that because the `Expecter` interface is so verbose in the generated code, that cluttering the tests with expected code would be too much. But I think that for this case, we really should have the expected code as a raw string in the test.

Reply:
I think we should stick with how the tests are done for now (strings directly in the test), but definitely look for improvements later on. As you mentioned, asserting mockery with mockeries output isn't correct, but we definitely can improve the readability of the tests, maybe even the way @Gevrai described. But it definitely is out of the scope of this, and it's a bigger decision that should be done correctly (so that we don't face it again in a year, hopefully :D ).

Reply:
I do really like the idea of keeping unit tests for the generated mocks (`mocks/pkg/fixtures/ExpecterTest_test.go`), that can probably stay. I think we should have more of these unit tests on the generated mocks as another layer.

Reply:
@Gevrai let me know if you could make the change (expected output as a string literal in the test). We can get it merged once that's settled. Thanks!

Reply:
Hey sorry! Things got in the way a bit, I'll have it done this evening.

Just to make sure
- Keep test in mocks/pkg/fixtures/ExpecterTest_test.go
- Inline generated expecter check instead of reading from file

Reply:
No worries thanks again for the contributions. Yeah those two points should be it!

Reply:
Done, should be good now.

FYI I've changed the name of the interface (ExpecterTest -> Expecter) for coherence with others, and added the expecter to be generated by Makefile so that the unit tests have something to run on!